plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -1280,68 +1294,70 @@ public boolean downloadToFile(String urlString, Plugin plugin, @CheckForNull Fil
      */
     @SuppressFBWarnings({"HTTP_PARAMETER_POLLUTION"})
     protected boolean downloadHttpToFile(String pluginUrl, Plugin plugin, File pluginFile, int maxRetries){
-        boolean success = false;
-        for (int i = 0; i < maxRetries; i++) {
-            success = true;
-            try {
-                if (pluginFile.exists()) {
-                    Files.delete(pluginFile.toPath());
-                }
-            } catch (IOException e) {
-                logVerbose(String.format("Unable to delete %s before retry %d", pluginFile, i + 1));
-            }
-            HttpClient httpClient = getHttpClient();
-            HttpClientContext context = HttpClientContext.create();
-            CredentialsProvider credentialsProvider = getCredentialsProvider();
-            if (credentialsProvider != null) {
-                context.setCredentialsProvider(credentialsProvider);
-            }
-            HttpGet httpGet = new HttpGet(pluginUrl);
-            try {
-                httpClient.execute(httpGet, new FileDownloadResponseHandler(pluginFile), context);
-            } catch (IOException e) {
-                String message = String.format("Unable to resolve plugin URL %s, or download plugin %s to file: %s",
-                pluginUrl, plugin.getName(), e.getMessage());
-                if (i >= maxRetries -1) {
-                    System.out.println(message);
-                } else {
-                    logVerbose(message);
-                }
-                success = false;
-            } finally {
-                // get final URI (after all redirects)
-                List<URI> locations = context.getRedirectLocations();
-                if (locations != null) {
-                    String message = String.format("%s %s from %s (attempt %d of %d)", success ? "Downloaded" : "Tried downloading", plugin.getName(), locations.get(locations.size() - 1), i+1, maxRetries);
-                    if (success) {
-                        logVerbose(message);
-                    } else {
-                        System.out.println(message);
-                    }
-                }
-            }
-
-            // Check if plugin is a proper ZIP file
-            if (success) {
-                try (JarFile ignored = new JarFile(pluginFile)) {
-                    plugin.setFile(pluginFile);
-                    logVerbose("Downloaded and validated plugin " + plugin.getName());
-                    break;
-                } catch (IOException e) {
-                    System.out.println("Downloaded file for " + plugin.getName() + " is not a valid ZIP");
-                    if (i >= maxRetries -1) {
-                        if (verbose) {
-                            e.printStackTrace();
-                        }
-                    }
-                    success = false;
-                }
-            }
-        }
-        return success;
-    }
-
-      /**
+    	try {
+	    	getViaHttpWithResponseHandler(
```

Comment:
can you fix the mixed tabs and spaces in this file that you changed please?

Reply:
done